### DISCUSSION TITLE
Finding local maxima of scalar attribute

### DISCUSSION THREAD

**Discussion by jpmaterial** [networkx/networkx#5075] [Q&A]
Is there any support for finding the nodes with local maxima of a scalar node attribute?

    G = nx.Graph()
    G.add_node( 0, t=5.0 )
    G.add_node( 1, t=3.5 )
    G.add_node( 2, t=4.0 )
    G.add_edges_from( [(0,1, (1,2))] )
    nx.find_local_node_maxima( G, 't' ) == [0,2]

**Comment by MridulS** [comment]
Maybe I am getting this wrong but this could be done with something like 
``` python
>>> max((attr['t'], n) for n, attr in G.nodes.data())
 (5.0, 0)
```

This will iterate through all the node attributes and fetch the attribute `t` and give you the node (`0`) with the maximal value of `t`.

**Reply by jpmaterial** [reply]
I'm looking for *local* minima, i.e. nodes which have higher values than all their neighbors. Note how in my above example node 2 is included in the solution, although its value for t is lower than the global maximum 5.

**Reply by MridulS** [reply]
Ahh yeah that makes more sense, something like this should probably be enough I think
``` python
In [9]: def find_local_node_maxima(G, attr):
   ...:     for n in G:
   ...:         max_neigh_attr = max(G.nodes[neigh][attr] for neigh in G.neighbors(n))
   ...:         if G.nodes[n][attr] > max_neigh_attr:
   ...:             yield n


In [10]: list(find_local_node_maxima(G, 't'))
Out[10]: [0, 2]
```
